operators/pkg/controller/common/defaults/pod_template.go

```diff
@@ -283,7 +283,7 @@ func (b *PodTemplateBuilder) WithInitContainers(initContainers ...corev1.Contain
 		}
 	}
 
-	b.PodTemplate.Spec.InitContainers = append(containers, b.PodTemplate.Spec.InitContainers...)
+	b.PodTemplate.Spec.InitContainers = append(b.PodTemplate.Spec.InitContainers, containers...)
```

Comment:
Why are we reversing the order here? Also the comment on the function is now out of sync with the code I believe.

Reply:
I got:
```bash
Initializing keystore.
+ echo 'Initializing keystore.'
+ /usr/share/elasticsearch/bin/elasticsearch-keystore create
bash: line 7: /usr/share/elasticsearch/bin/elasticsearch-keystore: No such file or directory
```

This is because I used `WithInitContainerDefaults()` that mounts `{"mountPath":"/usr/share/elasticsearch/bin","name":"elastic-internal-elasticsearch-bin-local"}
` that is not yet populated by the prepare fs init container.

So:
- I found it was more convenient to run the `init-keystore` init container after the `elastic-internal-init-filesystem` init container to use the regular path of the `elasticsearch-keystore` (`/usr/share/elasticsearch/bin`)
- I felt more natural to keep the insertion order: the last inserted element is the last element in the slice

Reply:
I think we want any user-provided init container to run *after* our prepare-fs init container. They probably need to be able to do things that will rely on directories set up in the prepare-fs init container. For example: installing additional plugins.

I think to your keystore issue you can simply swap the prepare-fs & keystore init container in the non-user provided init containers we add?